packages/editor/src/lib/editor/derivations/notVisibleShapes.ts

```diff
@@ -38,68 +33,17 @@ export const notVisibleShapes = (editor: Editor) => {
 		})
 		return notVisibleShapes
 	}
-	return computed<Set<TLShapeId>>('getCulledShapes', (prevValue, lastComputedEpoch) => {
+	return computed<Set<TLShapeId>>('getCulledShapes', (prevValue) => {
 		if (!isCullingOffScreenShapes) return new Set<TLShapeId>()
 
 		if (isUninitialized(prevValue)) {
 			return fromScratch(editor)
 		}
-		const diff = shapeHistory.getDiffSince(lastComputedEpoch)
 
-		if (diff === RESET_VALUE) {
-			return fromScratch(editor)
-		}
-
-		const currentPageId = editor.getCurrentPageId()
-		if (lastPageId !== currentPageId) {
-			return fromScratch(editor)
-		}
-		const viewportPageBounds = editor.getViewportPageBounds()
-		if (!prevViewportPageBounds || !viewportPageBounds.equals(prevViewportPageBounds)) {
-			return fromScratch(editor)
-		}
-
-		let nextValue = null as null | Set<TLShapeId>
-		const addId = (id: TLShapeId) => {
-			// Already added
-			if (prevValue.has(id)) return
-			if (!nextValue) nextValue = new Set(prevValue)
-			nextValue.add(id)
-		}
-		const deleteId = (id: TLShapeId) => {
-			// No need to delete since it's not there
-			if (!prevValue.has(id)) return
-			if (!nextValue) nextValue = new Set(prevValue)
-			nextValue.delete(id)
-		}
-
-		for (const changes of diff) {
-			for (const record of Object.values(changes.added)) {
-				if (isShape(record)) {
-					const isCulled = isShapeNotVisible(editor, record.id, viewportPageBounds)
-					if (isCulled) {
-						addId(record.id)
-					}
-				}
-			}
-
-			for (const [_from, to] of Object.values(changes.updated)) {
-				if (isShape(to)) {
-					const isCulled = isShapeNotVisible(editor, to.id, viewportPageBounds)
-					if (isCulled) {
-						addId(to.id)
-					} else {
-						deleteId(to.id)
-					}
-				}
-			}
-			for (const id of Object.keys(changes.removed)) {
-				if (isShapeId(id)) {
-					deleteId(id)
-				}
-			}
-		}
+		const nextValue = fromScratch(editor)
 
-		return nextValue ?? prevValue
+		const isSame =
+			prevValue.size === nextValue.size && [...prevValue].every((id) => nextValue.has(id))
```

Comment:
It would be faster I think to do 
`!arr.some(id => !nextValue.has(id))`. Same worst-case but better best-case speed.

Reply:
yes true. also we could avoid allocating a new array by iterating over the set 
```diff
-	const isSame =
-		prevValue.size === nextValue.size && [...prevValue].every((id) => nextValue.has(id))
-	return isSame ? prevValue : nextValue
+ if (prevValue.size !== nextValue.size) return nextValue
+ for (const prev of prevValue) {
+   if (!nextValue.has(prev)) return nextValue
+ }
+ return prevValue
```